pusher-platform-core/src/main/kotlin/com/pusher/platform/network/Futures.kt

```diff
@@ -2,8 +2,12 @@ package com.pusher.platform.network
 
 import java.util.concurrent.*
 import java.util.concurrent.TimeUnit.SECONDS
+import java.util.concurrent.atomic.AtomicInteger
 
-private val futuresExecutorService = Executors.newFixedThreadPool(Runtime.getRuntime().availableProcessors())
+private val threadCount = AtomicInteger()
+private val futuresExecutorService: ExecutorService = Executors.newCachedThreadPool{
+    Thread(it, "Pusher-Thread-${threadCount.getAndIncrement()}")
```

Comment:
Will this ever go above `0`?

Reply:
Indeed: https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/atomic/AtomicInteger.html#incrementAndGet()

Reply:
But what is the scope of the `futuresExecutorService`? How does a new instance of the `ExecutorService` get created? I'm not sure on the kotlin scoping, but I had assumed variables introduced outside of any scope like that were static...

Reply:
Oh, I see, sorry. There's only one `ExecutorService` but the anonymous argument is the thread factory and the executor pools multiple threads. Never mind.

Reply:
I can split them into two variables to make it easier to understand